volatility/framework/plugins/windows/netscan.py

```diff
@@ -0,0 +1,306 @@
+        return [
+            requirements.TranslationLayerRequirement(name = 'primary',
+                                                     description = 'Memory layer for the kernel',
+                                                     architectures = ["Intel32", "Intel64"]),
+            requirements.SymbolTableRequirement(name = "nt_symbols", description = "Windows kernel symbols"),
+            requirements.PluginRequirement(name = 'poolscanner', plugin = poolscanner.PoolScanner, version = (1, 0, 0)),
+            requirements.BooleanRequirement(name = 'include-corrupt',
+                description = "Radically eases result validation. This will show partially overwritten data. WARNING: the results are likely to include garbage and/or corrupt data. Be cautious!",
+                default = cls.CORRUPT_DEFAULT,
+                optional = True
+            ),
+        ]
+
+    @staticmethod
+    def create_netscan_constraints(context, symbol_table: str) -> List[poolscanner.PoolConstraint]:
+        """Creates a list of Pool Tag Constraints for network objects.
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            symbol_table: The name of an existing symbol table containing the symbols / types
+
+        Returns:
+            The list containing the built constraints.
+        """
+
+        tcpl_size = context.symbol_space.get_type(symbol_table + constants.BANG + "_TCP_LISTENER").size
+        tcpe_size = context.symbol_space.get_type(symbol_table + constants.BANG + "_TCP_ENDPOINT").size
+        udpa_size = context.symbol_space.get_type(symbol_table + constants.BANG + "_UDP_ENDPOINT").size
+
+        # ~ vollog.debug("Using pool size constraints: TcpL {}, TcpE {}, UdpA {}".format(tcpl_size, tcpe_size, udpa_size))
+
+        return [
+            # TCP listener
+            poolscanner.PoolConstraint(b'TcpL',
+                                       type_name = symbol_table + constants.BANG + "_TCP_LISTENER",
+                                       size = (tcpl_size, None),
+                                       page_type = poolscanner.PoolType.NONPAGED | poolscanner.PoolType.FREE),
+            # TCP Endpoint
+            poolscanner.PoolConstraint(b'TcpE',
+                                       type_name = symbol_table + constants.BANG + "_TCP_ENDPOINT",
+                                       size = (tcpe_size, None),
+                                       page_type = poolscanner.PoolType.NONPAGED | poolscanner.PoolType.FREE),
+            # UDP Endpoint
+            poolscanner.PoolConstraint(b'UdpA',
+                                       type_name = symbol_table + constants.BANG + "_UDP_ENDPOINT",
+                                       size = (udpa_size, None),
+                                       page_type = poolscanner.PoolType.NONPAGED | poolscanner.PoolType.FREE)
+        ]
+
+    def determine_tcpip_version(self) -> str:
```

Comment:
This method relies on `windows.info` being correct. All other approaches, including
- checking tcpip.sys's PDB GUID
- checking tcpip.sys in memory for its file version PE header
- dumping the tcpip.sys file from memory for the version PE header
have not worked reliably as those artifacts are either insufficient or got corrupted along the way.